course1/13.popular_package/7.package_os/task1.13.7.2/main.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+func WriteFile(data io.Reader, fd io.Writer) error {
+	if file, ok := fd.(*os.File); ok {
+		dir := filepath.Dir(file.Name())
+		err := os.MkdirAll(dir, 0755)
+		if err != nil {
+			return fmt.Errorf("MkdirAll: %s", err)
+		}
+	} else {
```

Comment:
you can write the same code without else
in Go it's preferable not to use alse at all
```
if !ok {
      return fmt.Errorf("not a file desc: %#v", fd)
}
```